modules/mongodb/charts.go

```diff
@@ -529,6 +571,46 @@ func (m *Mongo) removeDatabasesFromDBStatsCharts(newDatabases []string) {
 	}
 }
 
+// removeReplicaSetMember removes dimensions for not existing
+// replicate set members
+func (m *Mongo) removeReplicaSetMembers(newMembers []string) {
+	diff := sliceDiff(m.replSetMembers, newMembers)
+	for _, name := range diff {
+		dimID := replicationHeartbeatLatencyDimPrefix + name
+		chart := m.charts.Get(replicationHeartbeatLatency)
```

Comment:
I think we need to check if "dimID" is in the "m.replSetDimsEnabled[id]" and only then try to get chart/delete the dimension.

I see that [lastHeartbeatRecv](lastHeartbeatRecv) and [pingMs](https://docs.mongodb.com/manual/reference/command/replSetGetStatus/#mongodb-data-replSetGetStatus.members-n-.pingMs) are not available for all replica set members.

Something like

```go
	diff := sliceDiff(m.replSetMembers, newMembers)
	for _, name := range diff {
		for _, v := range []struct{ chartID, dimPrefix string }{
			{chartReplLag.ID, replicationLagDimPrefix},
			{chartReplHeartbeatLatency.ID, replicationHeartbeatLatencyDimPrefix},
			{chartReplPing.ID, replicationNodePingDimPrefix},
		} {
			id := v.dimPrefix + name
			if !m.replSetDimsEnabled[id] {
				continue
			}
            delete(m.replSetDimsEnabled, id)
...
```

Reply:
sounds good, I'll change it and also use the constants instead of the chart IDs. For example: `replicationLag` instead of `chartReplLag.ID`